client/state/data-layer/wpcom/timezones/index.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+
+import { TIMEZONES_REQUEST } from 'state/action-types';
+
+import {
+	timezonesRequestSuccessAction,
+	timezonesRequestFailureAction,
+	timezonesReceiveAction,
+} from 'state/timezones/actions';
+
+const undocumented = wpcom.undocumented();
+
+/*
+ * Start a request to WordPress.com server to get the timezones data
+ */
+export const requestTimezones = ( { dispatch } ) => (
+	undocumented
+		.timezones()
```

Comment:
Is this an existing function inside of `undocumented()`?

With the data middleware I'm strongly encouraging people to rely more on `wpcom.req.get()` than writing code inside of `wpcom.js` and then calling it.

As we move (hopefully very soon) to using the HTTP middleware this will be necessary in order to gain the benefits of things like error/retry policies and offline queuing.

That is to say that if we can centralize the API-specific code in the API-specific folder/sub-system and then do the same for HTTP things then we will win big.